Pythonwin/pywin/framework/scriptutils.py

```diff
@@ -608,16 +608,15 @@ def _HandlePythonFailure(what, syntaxErrorPathName=None):
     typ, details, tb = sys.exc_info()
     if isinstance(details, SyntaxError):
         try:
-            msg, (fileName, line, col, text) = details
+            filename = details.filename
             if (not fileName or fileName == "<string>") and syntaxErrorPathName:
                 fileName = syntaxErrorPathName
-            _JumpToPosition(fileName, line, col)
+            _JumpToPosition(filename, details.lineno, details.offset)
         except (TypeError, ValueError):
-            msg = str(details)
-        win32ui.SetStatusText("Failed to " + what + " - syntax error - %s" % msg)
+            pass
     else:
         traceback.print_exc()
-        win32ui.SetStatusText("Failed to " + what + " - " + str(details))
+    win32ui.SetStatusText(f"Failed to {what} - {type(details)} - {details}")
```

Comment:
I have also considered the following:
```suggestion
    win32ui.SetStatusText(f"Failed to {what} - {details!r}")
```